dev-docs/design/how-lit-works.md

```diff
@@ -0,0 +1,320 @@
+- `` html`<div>${document.createTextNode('hi')}</div>` ``, sets the Text node on the child part.
+
+At a very high level, `ChildPart` tracks the last value it was updated with via `_$committedValue`, skipping any work if the passed value is equal to the last committed value. This is how ChildParts diff changes at the value level.
+If the `value` being set on the ChildPart is not the same as `_$committedValue`, then the value is processed (differently based on the type) into a Node or Nodes and is committed to the DOM. Then the `value` is set on `_$committedValue` so the following `_$setValue` call can omit additional work if the value is unchanged.
+
+Primitive values set on a `ChildPart`, (`null`, `undefined`, `boolean`, `number`, `string`, `symbol`, or `bigint`), create or update a Text node. Committing a text node is done in `_commitText`, which is also where text sanitization occurrs.
+
+If the value is a `TemplateResult`, then ChildPart executes `_commitTemplateResult` which renders that `TemplateResult` into the ChildPart, [creating and updating](#3-create) that `TemplatResult`.
+
+If the value is a Node, the the ChildPart clears any previously rendered nodes and inserts the Node value directly.
+
+### AttributePart#\_$setValue
+
+Setting a value on an `AttributePart` occurs whenever the authored template has a value or values bound to an attribute without the prefixes (`.`, `@`, `?`).
+
+In the single binding within an attribute value case such as `` html`<input value=${...}>` ``, an AttributePart once again uses `_$committedValue` to not do extra work. But otherwise calls `this.element.setAttribute(name, value)`.
+
+An AttributePart also handles the more complex case where multiple bindings might be within an attribute: `` html`<div class="${...} static-class ${...}"></div>` ``, but again this results in a `setAttribute` call.
+
+### PropertyPart#\_$setValue
+
+This extends `AttributePart`, with the only difference being that committing the value to the live DOM is not `setAttribute(name, value)`, but instead as a property via `this.element[name] = value`.
+
+A PropertyPart is authored the same as an AttributePart, but the attribute name is prefixed with `.`, e.g.: `` html`<input .value=${...}>` ``.
+
+### BooleanAttributePart#\_$setValue
+
+BooleanAttributePart also extends `AttributePart`, and overrides `_commitValue`. When the value is committed to a BooleanAttributePart, it is committed with roughly `this.element.toggleAttribute(name, value)`. A BooleanAttributePart is authored the same as an AttributePart, but the attribute name is prefixed with `?`, e.g.: `` html`<input ?checked=${...}>` ``.
+
+### EventPart#\_$setValue
+
+EventPart also extends `AttributePart`, and overrides `_$setValue` to take the user provided event listener and then call `this.element.addEventListener(attributeName, value)`.
+The EventPart ensures that event listeners are added and cleaned up between renders and if the passed in value changes.
```

Comment:
Maybe worth adding that values of `null`, `undefined`, or `nothing` will remove any old listeners and not add any new listeners?

Reply:
Great suggestion! Done!